### PR TITLE
Clean up output and logs

### PR DESCRIPTION
**Pull request by pcapriotti**
- integration: Remove debug messages from ModService tests
- Do not log rabbit MQ connection failures on async exceptions
- cannon: Do not print uncaught SignalledToExit exceptions to stdout

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: libs/extended/src/Network/AMQP/Extended.hs [hunk 1]
```diff
@@ -65,12 +66,15 @@ openConnectionWithRetries l host port vHost hooks = do
...
-          logError willRetry e retryStatus = do
+          logError willRetry e retryStatus = when (isSync e) $ do
             Log.err l $
               Log.msg (Log.val "Failed to connect to RabbitMQ")
                 . Log.field "error" (displayException @SomeException e)
                 . Log.field "willRetry" willRetry
                 . Log.field "retryCount" retryStatus.rsIterNumber
+          isSync e = case E.fromException e of
+            Just (SomeAsyncException _) -> False
+            _ -> True
```
Comment (akshaymankar) [inline libs/extended/src/Network/AMQP/Extended.hs:77]: I think I made a mistake here, we're not supposed to recover from async exceptions. Could you please use [skipAsyncExceptions](https://hackage.haskell.org/package/retry-0.9.3.1/docs/Control-Retry.html#v:skipAsyncExceptions) before the `logRetries` handler in the `recovering` call?

**Review by akshaymankar — CHANGES_REQUESTED** [review]

File: libs/extended/src/Network/AMQP/Extended.hs (hunk 1 above)
Comment (pcapriotti) [reply libs/extended/src/Network/AMQP/Extended.hs:77]: Yeah, that makes sense. I was confused as to why we were retrying on async exceptions, but I was trying to keep the behaviour unchanged.

File: services/cannon/src/Cannon/Run.hs [hunk 2]
```diff
@@ -113,8 +117,14 @@ signalHandler e mainThread = CatchOnce $ do
   runWS e drain
   throwTo mainThread SignalledToExit
 
+uncaughtExceptionHandler :: SomeException -> IO ()
+uncaughtExceptionHandler e = do
+  case E.fromException e of
+    Just SignalledToExit -> pure () -- swallow
```
[the remark this answers is not in the document]
Comment (fisx) [reply services/cannon/src/Cannon/Run.hs:123]: does this rethrow?  if it swallows the exception, does it still terminate the main thread?

File: services/cannon/src/Cannon/Run.hs [hunk 3]
```diff
@@ -125,11 +135,11 @@ refreshMetrics = do
   safeForever $ do
     s <- D.size c
     gaugeSet (fromIntegral s) (path "net.websocket.clients") m
-    threadDelay 1000000
+    liftIO $ threadDelay 1000000
```
[the remark this answers is not in the document]
Comment (pcapriotti) [reply services/cannon/src/Cannon/Run.hs:138]: Before we were using `threadDelay` from `UnliftIO`. I removed the `UnliftIO` import to make sure we are using the normal `throwTo`, and not the one from `UnliftIO`, which is doing some unnecessary wrapping and making it hard to match on the exception in the handler. We could keep the `UnliftIO` import just for `threadDelay`, but that just saves a `liftIO`.

**Review by akshaymankar — APPROVED** [review]
:rocket:

**Pull request merged**